File: backend/app/schemas/emergency.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class EmergencyAnalysisResponse(BaseModel):
    id: str
    user_id: str
    report_id: Optional[str] = None
    tree_id: Optional[str] = None
    image_url: str
    analysis_status: str  # "completed", "unavailable", "inconclusive", "failed"
    is_ai_available: bool = False
    emergency_detected: bool = False
    severity: str = "None"  # "High", "Medium", "Low", "None", "Unknown"
    confidence: Optional[float] = 0.0
    detected_issue: str
    explanation: str
    recommended_action: str
    detected_conditions: List[str] = []
    risk_factors: List[str] = []
    location_name: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

class EmergencyAnalysisListResponse(BaseModel):
    items: List[EmergencyAnalysisResponse]
    total: int

File: src/services/emergency.ts
import { apiRequest } from './api';

export type EmergencyAnalysisStatus = 'completed' | 'unavailable' | 'inconclusive' | 'failed';
export type EmergencySeverity = 'High' | 'Medium' | 'Low' | 'None' | 'Unknown';

export interface EmergencyAnalysisResponse {
  id: string;
  user_id: string;
  report_id?: string | null;
  tree_id?: string | null;
  image_url: string;
  analysis_status: EmergencyAnalysisStatus;
  is_ai_available: boolean;
  emergency_detected: boolean;
  severity: EmergencySeverity;
  confidence: number;
  detected_issue: string;
  explanation: string;
  recommended_action: string;
  detected_conditions: string[];
  risk_factors: string[];
  location_name?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  created_at: string;
  updated_at?: string | null;
}

export const emergencyApi = {
  analyze: (formData: FormData): Promise<EmergencyAnalysisResponse> => {
    return apiRequest<EmergencyAnalysisResponse>('/api/emergency/analyze', {
      method: 'POST',
      body: formData,
    });
  },

  getMyAnalyses: (): Promise<EmergencyAnalysisResponse[]> => {
    return apiRequest<EmergencyAnalysisResponse[]>('/api/emergency/my', {
      method: 'GET',
    });
  },

  getAnalysisById: (analysisId: string): Promise<EmergencyAnalysisResponse> => {
    return apiRequest<EmergencyAnalysisResponse>(`/api/emergency/${encodeURIComponent(analysisId)}`, {
      method: 'GET',
    });
  },
};
